File: solutions/lua/space-age/1/space-age.lua
local SpaceAge = {}

function SpaceAge:new(n)
	local base = 31557600.0
	local age = {}
	age.seconds = n
	age["on_mercury"] = function() return n / (base * 0.2408467) end
	age["on_earth"] = function() return n / (base * 1) end
	age["on_venus"] = function() return n / (base * 0.61519726) end
	age["on_mars"] = function() return n / (base * 1.8808158) end
	age["on_jupiter"] = function() return n / (base * 11.862615) end
	age["on_saturn"] = function() return n / (base * 29.447498) end
	age["on_uranus"] = function() return n / (base * 84.016846) end
	age["on_neptune"] = function() return n / (base * 164.79132) end
	return age
end

return SpaceAge

File: solutions/c/space-age/1/space_age.c
#include "space_age.h"

const int base = 31557600.0;

#define ON(X, F) if (planet == X) return seconds / (base * F);

float age(planet_t planet, int64_t seconds) {
	ON(MERCURY, 0.2408467);
	ON(EARTH, 1.0);
	ON(VENUS, 0.61519726);
	ON(MARS, 1.8808158);
	ON(JUPITER, 11.862615);
	ON(SATURN, 29.447498);
	ON(URANUS, 84.016846);
	ON(NEPTUNE, 164.79132);
	return -1;
}
